src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -320,39 +324,17 @@ public SnowflakeStreamingIngestChannelInternal<?> openChannel(OpenChannelRequest
         getName());
 
     try {
-      Map<Object, Object> payload = new HashMap<>();
-      payload.put(
-          "request_id", this.flushService.getClientPrefix() + "_" + counter.getAndIncrement());
-      payload.put("channel", request.getChannelName());
-      payload.put("table", request.getTableName());
-      payload.put("database", request.getDBName());
-      payload.put("schema", request.getSchemaName());
-      payload.put("write_mode", Constants.WriteMode.CLOUD_STORAGE.name());
-      payload.put("role", this.role);
-      if (request.isOffsetTokenProvided()) {
-        payload.put("offset_token", request.getOffsetToken());
-      }
-
-      OpenChannelResponse response =
-          executeWithRetries(
-              OpenChannelResponse.class,
-              OPEN_CHANNEL_ENDPOINT,
-              payload,
-              "open channel",
-              STREAMING_OPEN_CHANNEL,
-              httpClient,
-              requestBuilder);
-
-      // Check for Snowflake specific response code
-      if (response.getStatusCode() != RESPONSE_SUCCESS) {
-        logger.logDebug(
-            "Open channel request failed, channel={}, table={}, client={}, message={}",
-            request.getChannelName(),
-            request.getFullyQualifiedTableName(),
-            getName(),
-            response.getMessage());
-        throw new SFException(ErrorCode.OPEN_CHANNEL_FAILURE, response.getMessage());
-      }
+      OpenChannelRequestInternal openChannelRequest =
+          new OpenChannelRequestInternal(
+              this.storageManager.getClientPrefix() + "_" + counter.getAndIncrement(),
+              this.role,
+              request.getDBName(),
+              request.getSchemaName(),
+              request.getTableName(),
+              request.getChannelName(),
+              Constants.WriteMode.CLOUD_STORAGE,
+              request.getOffsetToken());
```

Comment:
`request.isOffsetTokenProvided() ? request.getOffsetToken() : null`

Reply:
Discussed offline. Keep ` @JsonInclude(JsonInclude.Include.NON_NULL)` so no need to check.